From 0cd385af7e915ad38cf9d99b02726799408ebe36 Mon Sep 17 00:00:00 2001
From: Anup Patel <anup.patel@wdc.com>
Date: Thu, 20 Jun 2019 11:53:32 +0530
Subject: [PATCH 15/21] doc: sifive-fu540: Update README for steps to create
 FW_PAYLOAD

Due changes in DT bindings, we now embed DTB from Linux-5.3 (or
higher) in OpenSBI FW_PAYLOAD along with payload u-boot.bin. This
patch updates SiFive FU540 README to reflect the changes in build
and boot steps.

Signed-off-by: Anup Patel <anup.patel@wdc.com>
Reviewed-by: Bin Meng <bmeng.cn@gmail.com>
Reviewed-by: Alistair Francis <Alistair.Francis@wdc.com>
Upstream-Status: Submitted
---
 doc/README.sifive-fu540 | 356 ++++++++++++++++++----------------------
 1 file changed, 164 insertions(+), 192 deletions(-)

diff --git a/doc/README.sifive-fu540 b/doc/README.sifive-fu540
index fd9f2a8e46..33e03dc861 100644
--- a/doc/README.sifive-fu540
+++ b/doc/README.sifive-fu540
@@ -13,7 +13,8 @@ The support for following drivers are already enabled:
 3. Cadence MACB ethernet driver for networking support.
 
 TODO:
-1. SPI driver is still missing. So MMC card can't be used in U-Boot as of now.
+1. SPI host driver is still missing.
+2. SPI MMC driver does not compile and needs a re-write using U-Boot DM.
 2. U-Boot expects the serial console device entry to be present under /chosen
    DT node. Example:
    chosen {
@@ -33,16 +34,21 @@ Building
 
 Flashing
 ========
-The current U-Boot port is supported in S-mode only and loaded from DRAM.
+The current U-Boot port is supported in S-mode only and loaded directly
+into DRAM.
 
-A prior stage (M-mode) firmware/bootloader (e.g OpenSBI or BBL) is required to
-load the u-boot.bin into memory and provide runtime services. The u-boot.bin
-can be given as a payload to the prior stage (M-mode) firmware/bootloader.
+A prior stage (M-mode) firmware/bootloader (e.g OpenSBI) is required to
+boot the u-boot.bin in S-mode and provide M-mode runtime services.
 
-The description of steps required to build the firmware is beyond the scope of
-this document. Please refer OpenSBI or BBL documenation.
+Currently, the u-boot.bin is used as a payload of the OpenSBI FW_PAYLOAD
+firmware. We need to compile OpenSBI with below command:
+make PLATFORM=sifive/fu540 FW_PAYLOAD_PATH=<path to u-boot.bin> FW_PAYLOAD_FDT_PATH=<path to hifive-unleashed-a00.dtb from Linux>
+(Note: Prefer hifive-unleashed-a00.dtb from Linux-5.3 or higher)
+(Note: Linux-5.2 is also fine but it does not have ethernet DT node)
+
+More detailed description of steps required to build FW_PAYLOAD firmware
+is beyond the scope of this document. Please refer OpenSBI documenation.
 (Note: OpenSBI git repo is at https://github.com/riscv/opensbi.git)
-(Note: BBL git repo is at https://github.com/riscv/riscv-pk.git)
 
 Once the prior stage firmware/bootloader binary is generated, it should be
 copied to the first partition of the sdcard.
@@ -55,20 +61,18 @@ Once you plugin the sdcard and power up, you should see the U-Boot prompt.
 
 Sample boot log from HiFive Unleashed board
 ===========================================
-U-Boot 2019.01-00019-gc7953536-dirty (Jan 22 2019 - 11:05:40 -0800)
+U-Boot 2019.07-rc4-00013-g1837f893b0 (Jun 20 2019 - 11:08:48 +0530)
 
 CPU:   rv64imafdc
-Model: sifive,hifive-unleashed-a00
+Model: SiFive HiFive Unleashed A00
 DRAM:  8 GiB
 In:    serial@10010000
 Out:   serial@10010000
 Err:   serial@10010000
-Net:
-Warning: ethernet@10090000 (eth0) using random MAC address - b6:75:4d:48:50:94
-eth0: ethernet@10090000
+Net:   eth0: ethernet@10090000
 Hit any key to stop autoboot:  0
 => version
-U-Boot 2019.01-00019-gc7953536-dirty (Jan 22 2019 - 11:05:40 -0800)
+U-Boot 2019.07-rc4-00013-g1837f893b0 (Jun 20 2019 - 11:08:48 +0530)
 
 riscv64-linux-gcc.br_real (Buildroot 2018.11-rc2-00003-ga0787e9) 8.2.0
 GNU ld (GNU Binutils) 2.31.1
@@ -79,30 +83,19 @@ Now you can configure your networking, tftp server and use tftp boot method to
 load uImage.
 
 ==========================================================================
-=> setenv ethaddr 70:B3:D5:92:F0:C2
-=> setenv ipaddr 10.196.157.189
-=> setenv serverip 10.11.143.218
-=> setenv gatewayip 10.196.156.1
+=> setenv ipaddr 10.206.5.241
 => setenv netmask 255.255.252.0
-=> bdinfo
-boot_params = 0x0000000000000000
-DRAM bank   = 0x0000000000000000
--> start    = 0x0000000080000000
--> size     = 0x0000000200000000
-relocaddr   = 0x00000000fff90000
-reloc off   = 0x000000007fd90000
-ethaddr     = 70:B3:D5:92:F0:C2
-IP addr     = 10.196.157.189
-baudrate    = 115200 bps
-=> tftpboot uImage
+=> setenv serverip 10.206.4.143
+=> setenv gateway 10.206.4.1
+=> tftpboot ${kernel_addr_r} /sifive/fu540/uImage
 ethernet@10090000: PHY present at 0
 ethernet@10090000: Starting autonegotiation...
 ethernet@10090000: Autonegotiation complete
-ethernet@10090000: link up, 1000Mbps full-duplex (lpa: 0x3800)
+ethernet@10090000: link up, 1000Mbps full-duplex (lpa: 0x7c00)
 Using ethernet@10090000 device
-TFTP from server 10.11.143.218; our IP address is 10.196.157.189; sending through gateway 10.196.156.1
-Filename 'uImage'.
-Load address: 0x80200000
+TFTP from server 10.206.4.143; our IP address is 10.206.5.241
+Filename '/sifive/fu540/uImage'.
+Load address: 0x80600000
 Loading: #################################################################
          #################################################################
          #################################################################
@@ -112,192 +105,171 @@ Loading: #################################################################
          #################################################################
          #################################################################
          #################################################################
+         ########################################
+         1.5 MiB/s
+done
+Bytes transferred = 9162364 (8bce7c hex)
+=> tftpboot ${ramdisk_addr_r} /sifive/fu540/uRamdisk
+ethernet@10090000: PHY present at 0
+ethernet@10090000: Starting autonegotiation...
+ethernet@10090000: Autonegotiation complete
+ethernet@10090000: link up, 1000Mbps full-duplex (lpa: 0x7c00)
+Using ethernet@10090000 device
+TFTP from server 10.206.4.143; our IP address is 10.206.5.241
+Filename '/sifive/fu540/uRamdisk'.
+Load address: 0x82500000
+Loading: #################################################################
          #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         #################################################################
-         ##########################################################
-         2.5 MiB/s
+         ##################################
+         448.2 KiB/s
 done
-Bytes transferred = 14939132 (e3f3fc hex)
-=> bootm 0x80200000 - 0x82200000
-## Booting kernel from Legacy Image at 80200000 ...
+Bytes transferred = 2398272 (249840 hex)
+=> setenv bootargs "root=/dev/ram rw console=ttySIF0 earlycon=sbi"
+=> bootm ${kernel_addr_r} ${ramdisk_addr_r} ${fdtcontroladdr}
+## Booting kernel from Legacy Image at 80600000 ...
    Image Name:   Linux
    Image Type:   RISC-V Linux Kernel Image (uncompressed)
-   Data Size:    14939068 Bytes = 14.2 MiB
+   Data Size:    9162300 Bytes = 8.7 MiB
    Load Address: 80200000
    Entry Point:  80200000
    Verifying Checksum ... OK
-## Flattened Device Tree blob at 82200000
-   Booting using the fdt blob at 0x82200000
+## Loading init Ramdisk from Legacy Image at 82500000 ...
+   Image Name:   Linux RootFS
+   Image Type:   RISC-V Linux RAMDisk Image (uncompressed)
+   Data Size:    2398208 Bytes = 2.3 MiB
+   Load Address: 00000000
+   Entry Point:  00000000
+   Verifying Checksum ... OK
+## Flattened Device Tree blob at ff795730
+   Booting using the fdt blob at 0xff795730
    Loading Kernel Image ... OK
-   Using Device Tree in place at 0000000082200000, end 0000000082205c69
+   Using Device Tree in place at 00000000ff795730, end 00000000ff799dac
 
 Starting kernel ...
 
 [    0.000000] OF: fdt: Ignoring memory range 0x80000000 - 0x80200000
-[    0.000000] Linux version 5.0.0-rc1-00020-g4b51f736 (atish@jedi-01) (gcc version 7.2.0 (GCC)) #262 SMP Mon Jan 21 17:39:27 PST 2019
-[    0.000000] initrd not found or empty - disabling initrd
+[    0.000000] Linux version 5.2.0-rc1-00003-gb9543e66e700 (anup@anup-lab-machine) (gcc version 8.2.0 (Buildroot 2018.11-rc2-00003-ga0787e9)) #1 SMP Thu Jun 20 11:41:26 IST 2019
+[    0.000000] earlycon: sbi0 at I/O port 0x0 (options '')
+[    0.000000] printk: bootconsole [sbi0] enabled
+[    0.000000] Initial ramdisk at: 0x(____ptrval____) (2398208 bytes)
 [    0.000000] Zone ranges:
 [    0.000000]   DMA32    [mem 0x0000000080200000-0x00000000ffffffff]
-[    0.000000]   Normal   [mem 0x0000000100000000-0x000027ffffffffff]
+[    0.000000]   Normal   [mem 0x0000000100000000-0x000000027fffffff]
 [    0.000000] Movable zone start for each node
 [    0.000000] Early memory node ranges
 [    0.000000]   node   0: [mem 0x0000000080200000-0x000000027fffffff]
 [    0.000000] Initmem setup node 0 [mem 0x0000000080200000-0x000000027fffffff]
-[    0.000000] software IO TLB: mapped [mem 0xfbfff000-0xfffff000] (64MB)
-[    0.000000] CPU with hartid=0 has a non-okay status of "masked"
-[    0.000000] CPU with hartid=0 has a non-okay status of "masked"
+[    0.000000] software IO TLB: mapped [mem 0xfb795000-0xff795000] (64MB)
+[    0.000000] CPU with hartid=0 is not available
+[    0.000000] CPU with hartid=0 is not available
 [    0.000000] elf_hwcap is 0x112d
-[    0.000000] percpu: Embedded 15 pages/cpu @(____ptrval____) s29720 r0 d31720 u61440
+[    0.000000] percpu: Embedded 17 pages/cpu s29592 r8192 d31848 u69632
 [    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 2067975
-[    0.000000] Kernel command line: earlyprintk
+[    0.000000] Kernel command line: root=/dev/ram rw console=ttySIF0 earlycon=sbi
 [    0.000000] Dentry cache hash table entries: 1048576 (order: 11, 8388608 bytes)
 [    0.000000] Inode-cache hash table entries: 524288 (order: 10, 4194304 bytes)
 [    0.000000] Sorting __ex_table...
-[    0.000000] Memory: 8178760K/8386560K available (3309K kernel code, 248K rwdata, 872K rodata, 9381K init, 763K bss, 207800K reserved, 0K cma-reserved)
+[    0.000000] Memory: 8182056K/8386560K available (5753K kernel code, 357K rwdata, 1804K rodata, 204K init, 808K bss, 204504K reserved, 0K cma-reserved)
 [    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
 [    0.000000] rcu: Hierarchical RCU implementation.
-[    0.000000] rcu:     RCU event tracing is enabled.
 [    0.000000] rcu:     RCU restricting CPUs from NR_CPUS=8 to nr_cpu_ids=4.
-[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 10 jiffies.
+[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 25 jiffies.
 [    0.000000] rcu: Adjusting geometry for rcu_fanout_leaf=16, nr_cpu_ids=4
 [    0.000000] NR_IRQS: 0, nr_irqs: 0, preallocated irqs: 0
-[    0.000000] plic: mapped 53 interrupts to 4 (out of 9) handlers.
-[    0.000000] riscv_timer_init_dt: Registering clocksource cpuid [0] hartid [1]
+[    0.000000] plic: mapped 53 interrupts with 4 handlers for 9 contexts.
+[    0.000000] riscv_timer_init_dt: Registering clocksource cpuid [0] hartid [2]
 [    0.000000] clocksource: riscv_clocksource: mask: 0xffffffffffffffff max_cycles: 0x1d854df40, max_idle_ns: 3526361616960 ns
-[    0.000008] sched_clock: 64 bits at 1000kHz, resolution 1000ns, wraps every 2199023255500ns
-[    0.000221] Console: colour dummy device 80x25
-[    0.000902] printk: console [tty0] enabled
-[    0.000963] Calibrating delay loop (skipped), value calculated using timer frequency.. 2.00 BogoMIPS (lpj=10000)
-[    0.001034] pid_max: default: 32768 minimum: 301
-[    0.001541] Mount-cache hash table entries: 16384 (order: 5, 131072 bytes)
-[    0.001912] Mountpoint-cache hash table entries: 16384 (order: 5, 131072 bytes)
-[    0.003542] rcu: Hierarchical SRCU implementation.
-[    0.004347] smp: Bringing up secondary CPUs ...
-[    1.040259] CPU1: failed to come online
-[    2.080483] CPU2: failed to come online
-[    3.120699] CPU3: failed to come online
-[    3.120765] smp: Brought up 1 node, 1 CPU
-[    3.121923] devtmpfs: initialized
-[    3.124649] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
-[    3.124727] futex hash table entries: 1024 (order: 4, 65536 bytes)
-[    3.125346] random: get_random_u32 called from bucket_table_alloc+0x72/0x172 with crng_init=0
-[    3.125578] NET: Registered protocol family 16
-[    3.126400] sifive-u54-prci 10000000.prci: Registered U54 core clocks
-[    3.126649] sifive-gemgxl-mgmt 100a0000.cadence-gemgxl-mgmt: Registered clock switch 'cadence-gemgxl-mgmt'
-[    3.135572] vgaarb: loaded
-[    3.135858] SCSI subsystem initialized
-[    3.136193] usbcore: registered new interface driver usbfs
-[    3.136266] usbcore: registered new interface driver hub
-[    3.136348] usbcore: registered new device driver usb
-[    3.136446] pps_core: LinuxPPS API ver. 1 registered
-[    3.136484] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
-[    3.136575] PTP clock support registered
-[    3.137256] clocksource: Switched to clocksource riscv_clocksource
-[    3.142711] NET: Registered protocol family 2
-[    3.143322] tcp_listen_portaddr_hash hash table entries: 4096 (order: 4, 65536 bytes)
-[    3.143634] TCP established hash table entries: 65536 (order: 7, 524288 bytes)
-[    3.145799] TCP bind hash table entries: 65536 (order: 8, 1048576 bytes)
-[    3.149121] TCP: Hash tables configured (established 65536 bind 65536)
-[    3.149591] UDP hash table entries: 4096 (order: 5, 131072 bytes)
-[    3.150094] UDP-Lite hash table entries: 4096 (order: 5, 131072 bytes)
-[    3.150781] NET: Registered protocol family 1
-[    3.230693] workingset: timestamp_bits=62 max_order=21 bucket_order=0
-[    3.241224] io scheduler mq-deadline registered
-[    3.241269] io scheduler kyber registered
-[    3.242143] sifive_gpio 10060000.gpio: SiFive GPIO chip registered 16 GPIOs
-[    3.242357] pwm-sifivem 10020000.pwm: Unable to find controller clock
-[    3.242439] pwm-sifivem 10021000.pwm: Unable to find controller clock
-[    3.243228] xilinx-pcie 2000000000.pci: PCIe Link is DOWN
-[    3.243289] xilinx-pcie 2000000000.pci: host bridge /soc/pci@2000000000 ranges:
-[    3.243360] xilinx-pcie 2000000000.pci:   No bus range found for /soc/pci@2000000000, using [bus 00-ff]
-[    3.243447] xilinx-pcie 2000000000.pci:   MEM 0x40000000..0x5fffffff -> 0x40000000
-[    3.243591] xilinx-pcie 2000000000.pci: PCI host bridge to bus 0000:00
-[    3.243636] pci_bus 0000:00: root bus resource [bus 00-ff]
-[    3.243676] pci_bus 0000:00: root bus resource [mem 0x40000000-0x5fffffff]
-[    3.276547] Serial: 8250/16550 driver, 4 ports, IRQ sharing disabled
-[    3.277689] 10010000.serial: ttySIF0 at MMIO 0x10010000 (irq = 39, base_baud = 0) is a SiFive UART v0
-[    3.786963] printk: console [ttySIF0] enabled
-[    3.791504] 10011000.serial: ttySIF1 at MMIO 0x10011000 (irq = 40, base_baud = 0) is a SiFive UART v0
-[    3.801251] sifive_spi 10040000.spi: mapped; irq=41, cs=1
-[    3.806362] m25p80 spi0.0: unrecognized JEDEC id bytes: 9d, 70, 19
-[    3.812084] m25p80: probe of spi0.0 failed with error -2
-[    3.817453] sifive_spi 10041000.spi: mapped; irq=42, cs=4
-[    3.823027] sifive_spi 10050000.spi: mapped; irq=43, cs=1
-[    3.828604] libphy: Fixed MDIO Bus: probed
-[    3.832623] macb: GEM doesn't support hardware ptp.
-[    3.837196] libphy: MACB_mii_bus: probed
-[    4.041156] Microsemi VSC8541 SyncE 10090000.ethernet-ffffffff:00: attached PHY driver [Microsemi VSC8541 SyncE] (mii_bus:phy_addr=10090000.ethernet-ffffffff:00, irq=POLL)
-[    4.055779] macb 10090000.ethernet eth0: Cadence GEM rev 0x10070109 at 0x10090000 irq 12 (70:b3:d5:92:f0:c2)
-[    4.065780] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
-[    4.072033] ehci-pci: EHCI PCI platform driver
-[    4.076521] usbcore: registered new interface driver usb-storage
-[    4.082843] softdog: initialized. soft_noboot=0 soft_margin=60 sec soft_panic=0 (nowayout=0)
-[    4.127465] mmc_spi spi2.0: SD/MMC host mmc0, no DMA, no WP, no poweroff
-[    4.133645] usbcore: registered new interface driver usbhid
-[    4.138980] usbhid: USB HID core driver
-[    4.143017] NET: Registered protocol family 17
-[    4.147885] pwm-sifivem 10020000.pwm: SiFive PWM chip registered 4 PWMs
-[    4.153945] pwm-sifivem 10021000.pwm: SiFive PWM chip registered 4 PWMs
-[    4.186407] Freeing unused kernel memory: 9380K
-[    4.190224] This architecture does not have kernel memory protection.
-[    4.196609] Run /init as init process
-Starting logging: OK
-Starting mdev...
-[    4.303785] mmc0: host does not support reading read-only switch, assuming write-enable
-[    4.311109] mmc0: new SDHC card on SPI
-[    4.317103] mmcblk0: mmc0:0000 SS08G 7.40 GiB
-[    4.386471]  mmcblk0: p1 p2
-sort: /sys/devices/platform/Fixed: No such file or directory
-modprobe: can't change directory to '/lib/modules': No such file or directory
-Initializing random[    4.759075] random: dd: uninitialized urandom read (512 bytes read)
- number generator... done.
-Starting network...
-udhcpc (v1.24.2) started
-Sending discover...
-Sending discover...
-[    7.927510] macb 10090000.ethernet eth0: link up (1000/Full)
-Sending discover...
-Sending select for 10.196.157.190...
-Lease of 10.196.157.190 obtained, lease time 499743
-deleting routers
-adding dns 10.86.1.1
-adding dns 10.86.2.1
-/etc/init.d/S50dropbear
-Starting dropbear sshd: [   12.772393] random: dropbear: uninitialized urandom read (32 bytes read)
-OK
+[    0.000007] sched_clock: 64 bits at 1000kHz, resolution 1000ns, wraps every 2199023255500ns
+[    0.008553] Console: colour dummy device 80x25
+[    0.012990] Calibrating delay loop (skipped), value calculated using timer frequency.. 2.00 BogoMIPS (lpj=4000)
+[    0.023103] pid_max: default: 32768 minimum: 301
+[    0.028269] Mount-cache hash table entries: 16384 (order: 5, 131072 bytes)
+[    0.035068] Mountpoint-cache hash table entries: 16384 (order: 5, 131072 bytes)
+[    0.042770] *** VALIDATE proc ***
+[    0.045610] *** VALIDATE cgroup1 ***
+[    0.049157] *** VALIDATE cgroup2 ***
+[    0.053743] rcu: Hierarchical SRCU implementation.
+[    0.058297] smp: Bringing up secondary CPUs ...
+[    0.064134] smp: Brought up 1 node, 4 CPUs
+[    0.069114] devtmpfs: initialized
+[    0.073281] random: get_random_u32 called from bucket_table_alloc.isra.10+0x4e/0x160 with crng_init=0
+[    0.082157] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041785100000 ns
+[    0.091634] futex hash table entries: 1024 (order: 4, 65536 bytes)
+[    0.098480] NET: Registered protocol family 16
+[    0.114101] vgaarb: loaded
+[    0.116397] SCSI subsystem initialized
+[    0.120358] usbcore: registered new interface driver usbfs
+[    0.125541] usbcore: registered new interface driver hub
+[    0.130936] usbcore: registered new device driver usb
+[    0.136618] clocksource: Switched to clocksource riscv_clocksource
+[    0.148108] NET: Registered protocol family 2
+[    0.152358] tcp_listen_portaddr_hash hash table entries: 4096 (order: 4, 65536 bytes)
+[    0.159928] TCP established hash table entries: 65536 (order: 7, 524288 bytes)
+[    0.169027] TCP bind hash table entries: 65536 (order: 8, 1048576 bytes)
+[    0.178360] TCP: Hash tables configured (established 65536 bind 65536)
+[    0.184653] UDP hash table entries: 4096 (order: 5, 131072 bytes)
+[    0.190819] UDP-Lite hash table entries: 4096 (order: 5, 131072 bytes)
+[    0.197618] NET: Registered protocol family 1
+[    0.201892] RPC: Registered named UNIX socket transport module.
+[    0.207395] RPC: Registered udp transport module.
+[    0.212159] RPC: Registered tcp transport module.
+[    0.216940] RPC: Registered tcp NFSv4.1 backchannel transport module.
+[    0.223445] PCI: CLS 0 bytes, default 64
+[    0.227726] Unpacking initramfs...
+[    0.260556] Freeing initrd memory: 2336K
+[    0.264652] workingset: timestamp_bits=62 max_order=21 bucket_order=0
+[    0.278452] NFS: Registering the id_resolver key type
+[    0.282841] Key type id_resolver registered
+[    0.287067] Key type id_legacy registered
+[    0.291155] nfs4filelayout_init: NFSv4 File Layout Driver Registering...
+[    0.298299] NET: Registered protocol family 38
+[    0.302470] Block layer SCSI generic (bsg) driver version 0.4 loaded (major 254)
+[    0.309906] io scheduler mq-deadline registered
+[    0.314501] io scheduler kyber registered
+[    0.354134] Serial: 8250/16550 driver, 4 ports, IRQ sharing disabled
+[    0.360725] 10010000.serial: ttySIF0 at MMIO 0x10010000 (irq = 4, base_baud = 0) is a SiFive UART v0
+[    0.369191] printk: console [ttySIF0] enabled
+[    0.369191] printk: console [ttySIF0] enabled
+[    0.377938] printk: bootconsole [sbi0] disabled
+[    0.377938] printk: bootconsole [sbi0] disabled
+[    0.387298] 10011000.serial: ttySIF1 at MMIO 0x10011000 (irq = 1, base_baud = 0) is a SiFive UART v0
+[    0.396411] [drm] radeon kernel modesetting enabled.
+[    0.409818] loop: module loaded
+[    0.412606] libphy: Fixed MDIO Bus: probed
+[    0.416870] macb 10090000.ethernet: Registered clk switch 'sifive-gemgxl-mgmt'
+[    0.423570] macb: GEM doesn't support hardware ptp.
+[    0.428469] libphy: MACB_mii_bus: probed
+[    1.053009] Microsemi VSC8541 SyncE 10090000.ethernet-ffffffff:00: attached PHY driver [Microsemi VSC8541 SyncE] (mii_bus:phy_addr=10090000.ethernet-ffffffff:00, irq=POLL)
+[    1.067548] macb 10090000.ethernet eth0: Cadence GEM rev 0x10070109 at 0x10090000 irq 7 (70:b3:d5:92:f2:f3)
+[    1.077330] e1000e: Intel(R) PRO/1000 Network Driver - 3.2.6-k
+[    1.083069] e1000e: Copyright(c) 1999 - 2015 Intel Corporation.
+[    1.089061] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
+[    1.095485] ehci-pci: EHCI PCI platform driver
+[    1.099947] ehci-platform: EHCI generic platform driver
+[    1.105196] ohci_hcd: USB 1.1 'Open' Host Controller (OHCI) Driver
+[    1.111286] ohci-pci: OHCI PCI platform driver
+[    1.115742] ohci-platform: OHCI generic platform driver
+[    1.121142] usbcore: registered new interface driver uas
+[    1.126269] usbcore: registered new interface driver usb-storage
+[    1.132331] mousedev: PS/2 mouse device common for all mice
+[    1.137978] usbcore: registered new interface driver usbhid
+[    1.143325] usbhid: USB HID core driver
+[    1.148022] NET: Registered protocol family 10
+[    1.152609] Segment Routing with IPv6
+[    1.155571] sit: IPv6, IPv4 and MPLS over IPv4 tunneling driver
+[    1.161927] NET: Registered protocol family 17
+[    1.165907] Key type dns_resolver registered
+[    1.171694] Freeing unused kernel memory: 204K
+[    1.175375] This architecture does not have kernel memory protection.
+[    1.181792] Run /init as init process
+           _  _
+          | ||_|
+          | | _ ____  _   _  _  _
+          | || |  _ \| | | |\ \/ /
+          | || | | | | |_| |/    \
+          |_||_|_| |_|\____|\_/\_/
+
+               Busybox Rootfs
 
-Welcome to Buildroot
-buildroot login:
+Please press Enter to activate this console.
+/ #
-- 
2.22.0

